Objects/frameobject.c

```diff
@@ -27,6 +27,9 @@ frame_getlocals(PyFrameObject *f, void *closure)
     if (PyFrame_FastToLocalsWithError(f) < 0)
         return NULL;
     PyObject *locals = f->f_frame->f_locals;
+    if (locals == NULL) {
+        Py_RETURN_NONE;
```

Comment:
the return value is expected to be a `PyDictObject*` or `NULL`

`None` is unacceptable here.

Reply:
It should not be called with `f_locals == NULL`, and it's an invariant here.

Reply:
I think `assert` works here.